### PR TITLE
[CDAP-5135] Workaround to get correct program status from Twill 

### PR DESCRIPTION
**Pull request by anew**
…if program fails or gets killed.

Also a question: Does anybody know why the UI sill still show a status of FAILED even though I set the status to KILLED?

### CODE REVIEW

File: cdap-app-fabric/src/main/java/org/apache/twill/yarn/YarnTwillController.java [hunk 1]
```diff
@@ -311,4 +323,14 @@ public ResourceReport getResourceReport() {
     // in case the user calls this before starting, return null
     return (resourcesClient == null) ? null : resourcesClient.get();
   }
+
+  // begin change CDAP-5135
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/org/apache/twill/yarn/YarnTwillController.java:327]: nit: We should have a TODO to remove these comments.

**Comment by sreevatsanraman** [comment]
Minor nit and Can we add a unit test for this? LGTM otherwise

File: cdap-app-fabric/src/main/java/org/apache/twill/yarn/YarnTwillController.java (hunk 1 above)
Comment (anew) [inline cdap-app-fabric/src/main/java/org/apache/twill/yarn/YarnTwillController.java:327]: these comments are just to help see what is different from Apache Twill. There is already a TODO at the top to remove the entire class once the changes are made in Twill.

**Comment by anew** [comment]
Unit test, unfortunately, is not possible, because it is only for the YARN mode, and unit tests don't go through that code path.

**Comment by sreevatsanraman** [comment]
LGTM

**Comment by anew** [comment]
@sreevatsanraman I figured out why the status went to FAILED and not KILLED. Please see latest commit for small change to make that happen.

**Comment by anew** [comment]
Build running here: http://builds.cask.co/browse/CDAP-RUT1-1

**Comment by sreevatsanraman** [comment]
last commit looks good.

**Comment by anew** [comment]
Thanks @sreevatsanraman. Will wait for build, squash and merge.

**Pull request merged**